spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1324,6 +1335,7 @@ def handle_folding_range(self, response):
         ranges = response['params']
         folding_panel = self.panels.get(FoldingPanel)
         folding_panel.update_folding(ranges)
+        self.request_symbols()

```

Comment:
Please move this call to `process_diagnostics` (below `request_folding`) and update the comment above those calls.